pallets/did/src/did_details.rs

```diff
@@ -642,6 +639,30 @@ pub struct DidCreationDetails<T: Config> {
 	pub new_service_details: Vec<DidEndpoint<T>>,
 }
 
+impl<T: Config> DidCreationDetails<T> {
+	pub fn calculate_deposit(&self) -> BalanceOf<T> {
```

Comment:
Do we still need this if we have the same function for `DidDetails`?

Reply:
Is the heavy work you are referring to the one that checks the validity of the input? If that's the case, it does not matter much, since we take fees for that anyway, based on the size of the input, and we never return any surplus in case we finish early (e.g., if the user has not enough funds to pay for the deposit). Hence I would say it makes sense to avoid duplication and remove the code to calculate the deposit from the creation details, and simply call it on the created `DidDetails` inside the creation operation.
Alternatively, you could also simply create the details from the creation, calculate and check for the deposit, then run all the checks.
But as I said, the extrinsic is benchmarked depending on the input, so it does not really matter if we return early from the function or not.